app/models/docker_builder_service.rb

```diff
@@ -98,4 +102,13 @@ def send_after_notifications
     Samson::Hooks.fire(:after_docker_build, build)
     SseRailsEngine.send_event('builds', { type: 'finish', build: BuildSerializer.new(build, root: nil) })
   end
+
+  def registry_credentials
+    return nil unless ENV['DOCKER_REGISTRY_USER'].present? || ENV['DOCKER_REGISTRY_EMAIL'].present?
```

Comment:
Just checking: are we using an `OR` in here because the API requires one of the two, `username` or `email`, plus the password, for authentication? I mean, what happens if `DOCKER_REGISTRY_USER` was not provided but `DOCKER_REGISTRY_EMAIL` was? Will it go through?